packages/data-service/src/data-service.ts

```diff
@@ -2102,18 +2096,12 @@ export class DataServiceImpl extends EventEmitter implements DataService {
     }
 
     if (abortSignal.aborted) {
-      throw new DataServiceOperationCancelledError();
+      throw createCancelError();
```

Comment:
```suggestion
      throw abortSignal.reason ?? createCancelError();
```

Our Electron version isn’t quite there yet but hopefully is soon.

Reply:
If reason is just anything passed to the `abort` method call, should we rather do something like `createCancelError(abortSignal.reason)` that will keep the reason on our cancel error? Otherwise our `isCancelError` checks will start failing, or would this be a desired behavior?

Reply:
So … I think if at some point we’d abort with anything except an `AbortError`, then we should also reject with that error? I don’t know why we’d want to do that, tbh. This suggestion here was more about the fact that if `.reason` is available, it’s the right thing to pass, with the main effect right now just being that it has the best stack trace for indicating when the `abort()` call happened.

Reply:
Got it, thanks for sharing your thoughts on this!

> then we should also reject with that error

I guess the question I'm trying to find an answer for: is it still a cancellation event from Compass perspective and should we be able to identify it (like should aggregation builder break if at some point Compass or DE start cancelling in-flight agg requests with some sort of custom `reason`)? Especially as you can pass anything to the `reason`, not only errors